### DISCUSSION TITLE
Which terminals are supported

### DISCUSSION THREAD

**Discussion by techbech** [charmbracelet/bubbletea#312] [Q&A] [answered]
Hello everyone,

Is there any kind soul who could tell me which terminals are supported by Bubbletea or point me to some resource that holds such information?

This is necessary information for me to know to decide whether to use Bubbletea or not.

**Comment by muesli** [comment, marked as answer]
Hey @techbech,

this is a really good question indeed, and we will try and cover that topic in the documentation really soon!

Bubble Tea is using `termenv` to communicate with terminals and I've started a little support chart in its repo: https://github.com/muesli/termenv#optional-feature-support

Bubble Tea is not using all of the optional features however, and typically also gracefully falls back to reasonable defaults if a particular feature isn't supported by a terminal. From my memory, it's "fully" supporting alacritty, foot, Konsole, st, wezterm, xterm, Apple Terminal, iTerm, as well as all vte-based terminals including including Gnome Terminal, guake, Pantheon Terminal, Terminator, Tilix, XFCE Terminal.

Color schemes can't automatically be queried on rxvt, screen, tmux, Linux Console, Windows cmd or Windows Terminal, but it uses environment variables like `COLORFGBG` to detect reasonable defaults. Fully supported apart from that.

As far as colors are concerned, these terminals support the full RGB range: alacritty, foot, iTerm, kitty, Konsole, st, tmux, vte-based, wezterm, Windows Terminal. 8-bit (256) colors are supported by rxvt, screen, xterm, Apple Terminal. We detect available colors and automatically adapt to the closest available color on each terminal (if they're lacking true color support).

These are obviously non-exhaustive lists and I'm sure there are many more terminals out there Bubble Tea is compatible with, but I just haven't had the time to personally verify them. Let us know if we missed testing with your favorite terminal!

**Reply by techbech** [reply]
Thank you @muesli
That was a really helpful answer!

**Reply by meowgorithm** [reply]
Small note with regard to color: Bubble Tea itself doesn’t provide any notion of color or styling however most people use [Lip Gloss](https://github.com/charmbracelet/lipgloss) for style and layout, which leverages `termenv` for color precisely as described above.
